extension/observer/README.md

```diff
@@ -0,0 +1,9 @@
+# observers
+
+Observers are implemented as an extension to discover networked endpoints like a Kubernetes pod, Docker container, or local listening port. Other components can subscribe to an observer instance to be notified of endpoints coming and going.
+
+Currently the only component that uses observers is the [receiver_creator](../../receiver/receivercreator/README.md).
+
+## Current Observers
+
+* [k8sobserver](k8sobserver/README.md)
```

Comment:
Not sure how many observers there are expected to be in total, but I'd consider just listing the info here directly instead of having separate READMEs for each.

i.e. similar to https://github.com/open-telemetry/opentelemetry-collector/blob/master/extension/README.md

Reply:
Yea I noticed that shortly after posting this comment 😄